mongo/session.go

```diff
@@ -193,6 +193,10 @@ func (s *sessionImpl) WithTransaction(ctx context.Context, fn func(sessCtx Sessi
 			default:
 			}
 
+			// End if context has timed out or been canceled, as retrying has no chance of success.
+			if IsDeadlineExceededError(err) {
+				return res, err
```

Comment:
Is there any reason to not explicitly abort it at this point to let the server free up resources?

Reply:
At this point, I think `AbortTransaction` will already have been called on L187.

Reply:
Well, that's a race condition. It really depends on how important it is at this point. It might be small enough that we don't care.